src/mmw/js/src/core/itsiEmbedInterface.js

```diff
@@ -0,0 +1,27 @@
+"use strict";
+
+var _ = require('underscore'),
+    iframePhone = require('iframe-phone'),
+    EMBED_FLAG = 'itsi_embed',
+    QUERY_SUFFIX = EMBED_FLAG + '=true';
+
+var ItsiEmbedInterface = function() {
+    this.phone = new iframePhone.getIFrameEndpoint();
+
+    this.url = window.location.href + '?' + QUERY_SUFFIX;
+
+    this.setLearnerUrl = function(url) {
+        if (url) {
+            this.url = url + (url.indexOf('?') > 0 ? '&' : '?') + QUERY_SUFFIX;
+        }
+
+        this.phone.post('setLearnerUrl', this.url);
+    };
+
+    this.phone.addListener('getLearnerUrl', _.bind(this.setLearnerUrl, this));
```

Comment:
If parent frame (an activity in ITSI) requests URL with a `getLearnerUrl` message, execute `setLearnerUrl` in response.